test/functions.jl

```diff
@@ -67,7 +69,29 @@ using Statistics
                 names(first(eachslice(nda; dims=2))) ==
                 (:a, :c)
             )
+    end
+
+    @testset "eachcol, eachrow" begin
+        if VERSION < v"1.1-"
+            using NamedDims: eachrow, eachcol
         end
+        nda = NamedDimsArray([10 20; 31 40], (:x, :y))
+
+        @test names(first(eachcol(nda))) == (:x,)
+        @test names(first(eachrow(nda))) == (:y,)
+
+        @test_broken names(collect(eachcol(nda))) == (:y,)
+        @test_broken names(collect(eachrow(nda))) == (:y,)
```

Comment:
why are these `test_broken`?
If there's a good reason, let's open an issue to fix them, and comment a link to that issue here

Reply:
This doesn't seem like it is broken at all.
that is a normal vector of named vectors.

Reply:
I think the name of the generator dimension would ideally be passed along, as is now done for `[x^s for x in nda]`. But I couldn't see a way to make this work without https://github.com/JuliaLang/julia/pull/32310 .